DependencyInjection/HttplugExtension.php

```diff
@@ -206,4 +196,55 @@ private function configureAuthentication(ContainerBuilder $container, array $con
                 ->addArgument(new Reference($authServiceKey));
         }
     }
+
+    /**
+     * @param ContainerBuilder $container
+     * @param string           $name
+     * @param array            $arguments
+     */
+    private function configureClient(ContainerBuilder $container, $name, array $arguments)
+    {
+        $serviceId = 'httplug.client.'.$name;
+        $def = $container->register($serviceId, DummyClient::class);
+
+        if (empty($arguments['plugins'])) {
+            $def->setFactory([new Reference($arguments['factory']), 'createClient'])
+                ->addArgument($arguments['config']);
+        } else {
+            $def->setFactory('Http\HttplugBundle\ClientFactory\PluginClientFactory::createPluginClient')
+                ->addArgument(
+                    array_map(
+                        function ($id) {
+                            return new Reference($id);
+                        },
+                        $arguments['plugins']
+                    )
+                )
+                ->addArgument(new Reference($arguments['factory']))
+                ->addArgument($arguments['config']);
+        }
+
+
+        /*
+         * Decorate the client with clients from client-common
+         */
+
+        if ($arguments['flexible_client'] && $arguments['add_http_methods']) {
+            throw new \LogicException('A http client can\'t be decorated with both FlexibleHttpClient and HttpMethodsClient. Only one of the following options can be true. ("flexible_client", "add_http_methods")');
```

Comment:
this should be validated in the Configuration class. you can add a `validate()` section (or was it `validation()`?) on the level that has the two options.